grammar-tools/convert-grammar.php

```diff
@@ -0,0 +1,283 @@
+<?php
+
+require_once __DIR__ . '/../../wp-includes/mysql/class-mysql-lexer.php';
+
+const GRAMMAR_FILE = __DIR__ . '/../../wp-includes/mysql/mysql-grammar.php';
+
+// Convert the original MySQLParser.g4 grammar to a JSON format.
+// The grammar is also flattened and expanded to an ebnf-to-json-like format.
```

Comment:
Let's add a TODO comment to migrate it to a grammar-based g4 parser at one point.